test/acceptance/features/bindAppToServiceWithLabelSelector.feature

```diff
@@ -0,0 +1,111 @@
+                name: service-binding-a-s-c
+            spec:
+                bindAsFiles: false
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-a-s-c
+                application:
+                    labelSelector:
+                      matchLabels:
+                        app: gen-app-a-s-c
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then Service Binding "service-binding-a-s-c" is ready
+        And The application env var "BACKEND_USERNAME" has value "AzureDiamond"
+        And The application env var "BACKEND_PASSWORD" has value "hunter2"
+
+    Scenario: Attempt to bind an application with label selector and name to a service
+        Given Generic test application "gen-app-a-s-e" is running
+        * CustomResourceDefinition backends.stable.example.com is available
+        * The Secret is present
+            """
+            apiVersion: v1
+            kind: Secret
+            metadata:
+                name: backend-secret
+            stringData:
+                username: AzureDiamond
+                password: hunter2
+            """
+        And The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: service-a-s-e
+                annotations:
+                    service.binding: path={.status.data.dbCredentials},objectType=Secret,elementType=map
+            status:
+                data:
+                    dbCredentials: backend-secret
+            """
+        When Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: service-binding-a-s-e
+            spec:
+                bindAsFiles: false
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-a-s-e
+                application:
+                    name: gen-app-a-s-e
+                    labelSelector:
+                      matchLabels:
+                        app: gen-app-a-s-e
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "service-binding-a-s-e" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "service-binding-a-s-e" should be changed to "False"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").reason" of Service Binding "service-binding-a-s-e" should be changed to "ApplicationNotFound"
```

Comment:
IMHO - `ApplicationNotFound` is not correct reason - simply we do not allow name and label to cooexist - Hence, something like `InvalidApplicationReference` should be more appropriate.

Reply:
I need some more time to figure out how to introduce a new failure reason.